data_structures/src/data_request.rs

```diff
@@ -357,6 +356,30 @@ pub fn calculate_tally_change(
         + dr_output.commit_fee * (witnesses - commits_count)
 }
 
+pub fn calculate_witness_reward(
+    commits_count: usize,
+    reveals_count: usize,
+    honests_count: usize,
+    dr_output: &DataRequestOutput,
+) -> (u64, u64) {
+    if commits_count == 0 {
+        (0, 0)
+    } else if reveals_count == 0 {
+        (dr_output.collateral, 0)
+    } else {
+        let liars_count = reveals_count - honests_count;
+        let slashed_count = (commits_count - reveals_count + liars_count) as u64;
+        let honests_count = honests_count as u64;
+        let slashed_collateral_reward = dr_output.collateral * slashed_count / honests_count;
+        let slashed_collateral_reminder = (dr_output.collateral * slashed_count) % honests_count;
```

Comment:
```suggestion
        let slashed_collateral_remainer = (dr_output.collateral * slashed_count) % honests_count;
```

:laughing:

Reply:
It's not `reminder` nor `remainer` but `remainder`:
https://en.wikipedia.org/wiki/Remainder